modules/albo-installation.adoc

```diff
@@ -53,236 +52,173 @@ $ cat <<EOF > "${SCRATCH}/trust-policy.json"
-----
-NAME                                                             READY   STATUS    RESTARTS   AGE
-aws-load-balancer-controller-cluster-6ddf658785-pdp5d            1/1     Running   0          99s
-aws-load-balancer-operator-controller-manager-577d9ffcb9-w6zqn   2/2     Running   0          2m4s
-----
 
-[id="aws-load-balancer-operator-validating-the-deployment_{context}"]
-== Validating the deployment
-
-. Create a new project:
+. Create an AWS IAM policy for the AWS Load Balancer Controller.
+.. Download the appropriate IAM policy:
 +
 [source,terminal]
 ----
-$ oc new-project hello-world
+$ curl -o ${SCRATCH}/controller-permission-policy.json https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/v2.12.0/docs/install/iam_policy.json
 ----
-+
-. Deploy a hello world application:
+.. Create the permission policy for the Controller:
 +
 [source,terminal]
 ----
-$ oc new-app -n hello-world --image=docker.io/openshift/hello-openshift
+$ aws iam create-policy \
+    --region "$REGION" \
+    --policy-name aws-load-balancer-controller-policy \
+    --policy-document file://${SCRATCH}controller-permission-policy.json
 ----
 +
-. Configure a NodePort service for the AWS ALB to connect to:
+Take note of the Controller policy ARN in the output. This is referred to as the `$CONTROLLER_POLICY_ARN` for the remainder of this process.
+
+. Create an AWS IAM role for the AWS Load Balancer Controller:
+.. Create the trust policy for the Controller role:
 +
 [source,terminal]
 ----
-$ cat << EOF | oc apply -f -
-apiVersion: v1
-kind: Service
-metadata:
-  name: hello-openshift-nodeport
-  namespace: hello-world
-spec:
-  ports:
-    - port: 80
-      targetPort: 8080
-      protocol: TCP
-  type: NodePort
-  selector:
-    deployment: hello-openshift
+$ cat <<EOF > controller-trust-policy.json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+          "Federated": "${OIDC_ARN}"
+        },
+        "Action": "sts:AssumeRoleWithWebIdentity",
+        "Condition": {
+          "StringEquals": {
+            "${OIDC_ENDPOINT}:sub": "system:serviceaccount:aws-load-balancer-operator:aws-load-balancer-controller-cluster"
+            }
+        }
+      }
+    ]
+  }
 EOF
 ----
-+
-. Deploy an AWS ALB using the AWS Load Balancer Operator:
+. Create the Controller role using the trust policy:
 +
 [source,terminal]
 ----
-$ cat << EOF | oc apply -f -
-apiVersion: networking.k8s.io/v1
-kind: Ingress
-metadata:
-  name: hello-openshift-alb
-  namespace: hello-world
-  annotations:
-    alb.ingress.kubernetes.io/scheme: internet-facing
-spec:
-  ingressClassName: alb
-  rules:
-    - http:
-        paths:
-          - path: /
-            pathType: Exact
-            backend:
-              service:
-                name: hello-openshift-nodeport
-                port:
-                  number: 80
-EOF
+$ aws iam create-role --role-name ${CLUSTER_NAME}-albo-controller \
```

Comment:
Same as above, missing the / on the document file. Suggested change to be consistent with above:

```
CONTROLLER_ROLE_ARN=$(aws iam create-role --role-name "${CLUSTER_NAME}-albo-controller" \
  --assume-role-policy-document "file://${SCRATCH}/controller-trust-policy.json" \
  --query Role.Arn --output text)

echo ${CONTROLLER_ROLE_ARN}
```